recipes/vcfsamplecompare/meta.yaml

```diff
@@ -0,0 +1,45 @@
+{% set name = "vcfsamplecompare" %}
+{% set realname = "vcfSampleCompare" %}
+{% set version = "v2.004" %}
+about:
+  home: https://github.com/hepcat72/{{ realname }}
+  license: GNU
+  license_file: LICENSE
+  summary: This script sorts and (optionally) filters the rows/variants of a VCF
+    file (containing data for 2 or more samples) based on the differences in the
+    variant data between samples or sample groups. Degree of "difference" is
+    determined by either the best possible degree of separation of sample groups
+    by genotype calls or the difference in average allelic frequency of each
+    sample or sample group (with a gap size threshold). The pair of samples or
+    sample groups used to represent the difference for a variant row is the one
+    leading to the greatest difference in consistent genotype or average allelic
+    frequencies (i.e. observation ratios, e.g. AO/DP) of the same variant state.
+    If sample groups are not specified, the pair of samples leading to the
+    greatest difference is greedily discovered and chosen to represent the
+    variant/row.
+
+build:
+  number: 0
+package:
+  name: {{ name }}
+  version: {{ version }}
+requirements:
+  build:
+    - perl
```

Comment:
`perl` should go in the `host:` section, see https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#host

Reply:
OK.  I'm confused.  I added:

```
  host:
    - perl
```

`conda build` succeeds, but when I do a local `circleci build`, I get:

```
+ vcfSampleCompare.pl --version
/opt/conda/conda-bld/vcfsamplecompare_1538760954997/test_tmp/run_test.sh: line 7: vcfSampleCompare.pl: command not found
```

I did it twice.  Failed twice.  I took the host back out and it succeeded again.

My guess is that the perl path on the she-bang line at the top of the script is different on the "host" compared to what I have locally.  I had noticed that the tests in the meta.yaml file were using my manually installed version of the script instead of a conda-built one.  The Makefile.pl script should be configuring things to edit that line for the "user's" perl path (the one they used to run Makefile.PL on the line `perl Makefile.PL` in the build.sh script).

So what do you think is going on here?  Is there no install performed on the host or is it not configuring the she-bang line correctly?

Reply:
Or do I need to add something to the PATH from build.sh?

Reply:
OK, I've been looking at build.sh scripts containing `Makefile.PL` and it looks like I need to supply $PREFIX in some arguments to Makefile.PL...  That makes sense.  I'll give it a go and update if/when I get it figured out.

Reply:
I feel like I'm closer, but I must be supplying the wrong environment variable?

I now get this error:

```
+ vcfSampleCompare.pl --version
/Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/test_tmp/run_test.sh: /Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/vcfSampleCompare.pl: /Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_build: bad interpreter: No such file or directory
```

So at least it's *trying* to run a conda version of the script instead of using the one I had installed manually, but the perl interpreter path is incorrect?  Not sure why the entire interpreter path isn't shown, so I double-checked the she-bang line:

```
>head /Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/vcfSampleCompare.pl
#!/Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_build_env/bin/perl
```

The build directory was moved because of the failed build, but the perl path on the she-bang appears to have been valid: `/Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/build_prefix_moved_vcfsamplecompare-v2.005-pl526_0_osx-64/bin/perl`.  I can run it, NP.

I tried manually editing the she-bang to reflect the moved build dir, and it runs no problem:

```
>/Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/vcfSampleCompare.pl
vcfSampleCompare.pl -i <file*...>... [OPTIONS]

* <file*...>...       VCF input file.
  -o <sfx>            VCF outfile suffix (appended to -i).
  -u <sfx>            [STDOUT] Summary outfile suffix (appended to -i).
  -s <str ...>...     [any^] A group of sample names for difference comparisons.
                      ^ See --extended usage.
  -d <int,...>...     [all] Minimum number of samples to use in a group to
                      determine difference with its partner.
  -a <flt>            [0.6] Allelic frequency difference threshold [0-1].
  --no-g              Do not use genotype calls (use allelic frequency).
  --no-f              Do not filter variant rows whose sample groups do not
                      differ (enough).
  --no-w              Do not add as many samples to sample groups as possible.
  --help              Print general info and file formats.
  --extended [<cnt>]  Print detailed usage.

* Required.
```

So I don't know why I get the error:

`/Users/rleach/local/miniconda3/conda-bld/vcfsamplecompare_1538766140354/_build: bad interpreter: No such file or directory`

Perhaps the parsed perl path is getting truncated (as shown in the error)?  What do you think?

Reply:
*sigh* I'm stumped again. I can't seem to satisfy the setup on the host. I'm going to head over to gitter and ask some questions...

Reply:
OK, I finally figured it out.

Reply:
It wasn't obvious/clear to me from the doc that perl shouldn't also be in the 'build:' section, since I was using a perl call in the build.sh script. Lance caught that in short order. I banged my head against this for a couple days.